cirq-core/cirq/ops/raw_types_test.py

```diff
@@ -151,6 +151,30 @@ def test_op_validate():
         op2.validate_args([cirq.LineQid(1, 2), cirq.LineQid(1, 2)])
 
 
+def test_disable_op_validation():
+    q0, q1 = cirq.LineQubit.range(2)
+    h_op = cirq.H(q0)
+
+    # Fails normally.
+    with pytest.raises(ValueError, match='Wrong number'):
+        _ = cirq.H(q0, q1)
+    with pytest.raises(ValueError, match='Wrong number'):
+        h_op.validate_args([q0, q1])
+
+    # Passes, skipping validation.
+    cirq.__cirq_debug__.set(False)
+    op = cirq.H(q0, q1)
+    assert op.qubits == (q0, q1)
+    h_op.validate_args([q0, q1])
```

Comment:
I think it'd be nice to add a context manager for changing the `__cirq_debug__` flag and making that the default way of using it. This ensures that state is restored even if an exception happens, which makes it much easier to know the state of the flag when looking at code.
```python
@contextlib.contextmanager
def with_debug(value: bool) -> Iterator[None]:
    token = __cirq_debug__.set(value)
    try:
        yield
    finally:
        token.reset()
```
Then can use that in these tests:
```python
with cirq.with_debug(False):
    op = cirq.H(q0, q1)
    assert op.qubits == (q0, q1)
    h_op.valudate_args([q0, q1])
```
I think we should encourage using a mechanism like this and discourage setting `cirq.__cirq_debug__` manually (other than maybe for interactive use).

Reply:
I think adding the context manager is fine, but I'd still like to set the default value to `__debug__` so that users have a way to disable validations in existing codebase without major modifications in the code.  

What do you think?

Reply:
added a context manager, PTAL